documentation/smg/04_building_with_cmake.md

```diff
@@ -1,38 +1,151 @@
+```
+
+_(Note that Bash uses two dashes (`--`) whilst PowerShell uses one.)_
+
+Generally, the default build options will be sufficient,
+but there are several configurable parameters.
+Both scripts have a `help` flag which display the available actions:
+
+PowerShell:
+
+```powershell
+./tools/build_config/build.ps1 -help
+```
+
+Bash:
+
+```bash
+./tools/build_config/build.sh --help
+```
+
+## Using CMake directly
+
 CMake provides a way to generate build files across multiple platforms. It also
 ships with a script to aid in compiling Matlab mex libraries.
 
+The following process is roughly what the build scripts automate.
+
 1. After cloning the Horace repository, open the CMake GUI and select the root
 of the repository as the source.
+
 2. Select `<Horace Root>/build/` as the binary directory.
-3. Click configure. When the dialogue appears select your desired generator.
-You should pick a compiler that is compatible with your Matlab version, e.g.
-Matlab R2017b is not compatible with Visual Studio 2019, so choose Visual Studio 2017.
-4. Make sure to also select your platform/architecture, this should also match
-your Matlab version: if you installed a 64-bit version of Matlab choose x64 as
-the platform.
-5. Now click finish in the dialogue. CMake will find the Matlab compiler and
-various libraries.
-6. Now click generate. The build files should be generated inside the DLL
-directory. On Windows you can open the `<Horace Root>/build/Horace.sln` file in
-Visual Studio and build the targets. On linux you can `cd` into the
-`<Horace Root>/build/` directory and run `make` depending on the generator you
-selected.
-
-To do steps 3-6 using the command line you can use the following commands (for
+Create this directory if it does not exist.
+
+3. Click configure.
+When the dialogue appears select your desired generator.
+You should pick a compiler that is compatible with your Matlab version,
+see the [version compatibility matrix](./09_version_compatibility.md)
+(whilst the version compatibility matrix is a good guide,
+the mex libraries will likely work if built using newer compiler versions).
+
+4. Make sure to also select your platform/architecture,
+this should also match your Matlab version:
+if you installed a 64-bit version of Matlab choose x64 as the platform.
+
+5. Now click finish in the dialogue.
+CMake will find the Matlab compiler and various libraries.
+
+6. Now click generate.
+The build files should be generated inside the DLL directory.
+On Windows you can open the `<Horace Root>/build/Horace.sln` file in Visual
+Studio and build the targets.
+
+7. The built mex files will be written into `horace_core/DLL`.
+
+To do steps 2-6 using the command line you can use the following commands (on
 linux use `-G "Unix Makefiles"`):
 
 `$ cmake -S <Horace Root> -B <Horace Root>/build -G "Visual Studio 15 2017 Win64"`
 
-`$ cmake --build <Horace Root>/build`
+`$ cmake --build <Horace Root>/build --config Release`
+
```

Comment:
Could find $ confusing as the command line prompt - might be the shell variable $cmake oddly written? Maybe prefer >, >> or [prompt]>.